DataSynapseApi/ArmoniK.DevelopmentKit.GridServer.Client/Properties.cs

```diff
@@ -117,21 +118,34 @@ public class Properties
       Configuration = configuration;
 
       var sectionGrpc = configuration.GetSection(SectionGrpc).Exists() ? configuration.GetSection(SectionGrpc) : null;
-      ConnectionString  = sectionGrpc!.GetSection(SectionEndPoint).Exists() ? sectionGrpc![SectionEndPoint] : null;
-      ConfSSLValidation = !sectionGrpc!.GetSection(SectionSSlValidation).Exists() || sectionGrpc![SectionSSlValidation] != "disable";
 
-      CaCertFilePem = !string.IsNullOrEmpty(caCertPem) ? caCertPem : (sectionGrpc!.GetSection(SectionCaCert).Exists() ? sectionGrpc![SectionCaCert] : caCertPem);
 
-      ClientCertFilePem = !string.IsNullOrEmpty(clientCertFilePem)
-        ? clientCertFilePem
-        : (sectionGrpc!.GetSection(SectionClientCert).Exists() ? sectionGrpc![SectionClientCert] : null);
+      if (sectionGrpc != null)
+      {
+        ConnectionString  = sectionGrpc.GetSection(SectionEndPoint).Exists() ? sectionGrpc[SectionEndPoint] : null;
+        ConfSSLValidation = !sectionGrpc.GetSection(SectionSSlValidation).Exists() || sectionGrpc[SectionSSlValidation] != "disable";
 
-      ClientKeyFilePem = !string.IsNullOrEmpty(clientKeyFilePem)
-        ? clientKeyFilePem
-        : (sectionGrpc!.GetSection(SectionClientKey).Exists() ? sectionGrpc![SectionClientKey] : null);
+        if (sectionGrpc.GetSection(SeccionMTLS).Exists() && sectionGrpc[SeccionMTLS].ToLower() == "true")
+        {
+          CaCertFilePem     = sectionGrpc.GetSection(SectionCaCert).Exists() ? sectionGrpc[SectionCaCert] : null;
+          ClientCertFilePem = sectionGrpc.GetSection(SectionClientCert).Exists() ? sectionGrpc[SectionClientCert] : null;
+          ClientKeyFilePem  = sectionGrpc.GetSection(SectionClientKey).Exists() ? sectionGrpc[SectionClientKey] : null;
+        }
+      }
+
+      if (clientCertFilePem != null) ClientCertFilePem = clientCertFilePem;
```

Comment:
I am not sure to understand these line in relation to the ones above. I am afraid that the ClientCertFilePem will be modified again (compare to the value line 131).
Also to improve the readability it is better to use the variable in the same orders.